File: NumberSort.java
import java.util.ArrayList;

public class NumberSort extends CommandApp{
    private int evenCount = 0;
    private int oddCount = 0;

    public void run(int[] nA){


        int[] evenNumbers = new int[10];
        int[] oddNumbers = new int[10];

        int[] numbers = nA;

        int incr = 0;
        
        while((numbers.length) >= ((incr)+1)){

            int num = numbers[incr];

            if(num %2 == 0){
                //even
                evenNumbers[evenCount] = numbers[incr];
                evenCount++;
            }else{
                //odd
                oddNumbers[oddCount] = numbers[incr];
                oddCount++;
            }

            incr++;
        }

        System.out.println("Even Numbers:");
        for (int n : evenNumbers) {
            System.out.println(n);
        }

        System.out.println("Odd Numbers:");
        for (int n : oddNumbers) {
            System.out.println(n);
        }

        compareNumbers(evenNumbers,oddNumbers);
    }

    public void compareNumbers(int[] numberArray1, int[] numberArray2){
        System.out.println("--Comparing Numbers--");

        int[] largerArray;
        int[] smallerArray;

        if(evenCount > oddCount){
            largerArray = numberArray1;
            smallerArray = numberArray2;
        }else{
            largerArray = numberArray2;
            smallerArray = numberArray1;
        }

        for (int i = 0; i <= 9; i++) {
            int num1 = largerArray[i];
            int num2 = smallerArray[i];
            
            if(num1 == num2){
                System.out.println(num1+" == "+num2);
            }else{
                boolean larger = (num1>num2);

                if(larger){
                    System.out.println(num1 +" > "+ num2);
                }else{
                    System.out.println(num1 + " < "+ num2);
                }
            }
            
            
        }
    }
}

File: NumberSort.py
from CommandApp import CommandApp

class NumberSort(CommandApp):

    oddNumbers = []
    evenNumbers = []
    numbers = []

    def run(self,n):
        self.numbers = n

        for element in self.numbers:
            if element %2 != 0:
                #odd
                self.oddNumbers.append(element)
            else:
                #even
                self.evenNumbers.append(element)

        print("Odd Numbers: ")
        print(self.oddNumbers)
        print("Even Numbers: ")
        print(self.evenNumbers)
        self.compareNumbers(self.oddNumbers,self.evenNumbers)


    def compareNumbers(self,numberList1, numberList2):
        print("-- Comparing Numbers --")

        largerList = [];
        smallerList = [];

        if len(numberList1)>len(numberList2) :
            largerList = numberList1
            smallerList = numberList2
        else:
            largerList = numberList2
            smallerList = numberList1
        
        for i in range(len(largerList)):
            if i <= len(smallerList)-1:
                num1 = largerList[i]
                num2 = smallerList[i]
                larger = num1>num2

                if larger :
                    print(num1," > ",num2)
                else :
                    print(num1," < ",num2)

            else:
                num1 = largerList[i]
                print(num1)
                
